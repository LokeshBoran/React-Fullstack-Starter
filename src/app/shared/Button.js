import React from 'react';
import { Button } from 'material-ui';
import { withStyles } from 'material-ui/styles';
import {
  red, pink, purple,
  deepPurple, indigo, blue,
  lightBlue, cyan, teal,
  green, lightGreen, lime,
  yellow, amber, orange,
  deepOrange, brown, grey,
  blueGrey
} from 'material-ui/colors';
import classNames from 'classnames';

export const styles = theme => {
  const raisedTheme = (color, background) => ({
    color,
    background: background[500],
    '&:hover': {
      background: background[700]
    }
  });

  const flatTheme = (color, background = color) => ({
    color: color[500],
    '&:hover': {
      background: background[50]
    }
  });

  const { white, black } = theme.palette.common;

  return {
    root: {},

    raisedRed: raisedTheme(white, red),
    raisedPink: raisedTheme(white, pink),
    raisedPurple: raisedTheme(white, purple),
    raisedDeepPurple: raisedTheme(white, deepPurple),
    raisedIndigo: raisedTheme(white, indigo),
    raisedBlue: raisedTheme(white, blue),
    raisedLightBlue: raisedTheme(black, lightBlue),
    raisedCyan: raisedTheme(black, cyan),
    raisedTeal: raisedTheme(white, teal),
    raisedGreen: raisedTheme(black, green),
    raisedLightGreen: raisedTheme(black, lightGreen),
    raisedLime: raisedTheme(black, lime),
    raisedYellow: raisedTheme(black, yellow),
    raisedAmber: raisedTheme(black, amber),
    raisedOrange: raisedTheme(black, orange),
    raisedDeepOrange: raisedTheme(white, deepOrange),
    raisedBrown: raisedTheme(white, brown),
    raisedGrey: raisedTheme(black, grey),
    raisedBlueGrey: raisedTheme(white, blueGrey),

    flatRed: flatTheme(red),
    flatPink: flatTheme(pink),
    flatPurple: flatTheme(purple),
    flatDeepPurple: flatTheme(deepPurple),
    flatIndigo: flatTheme(indigo),
    flatBlue: flatTheme(blue),
    flatLightBlue: flatTheme(lightBlue),
    flatCyan: flatTheme(cyan),
    flatTeal: flatTheme(teal),
    flatGreen: flatTheme(green),
    flatLightGreen: flatTheme(lightGreen),
    flatLime: flatTheme(lime),
    flatYellow: flatTheme(yellow),
    flatAmber: flatTheme(amber),
    flatOrange: flatTheme(orange),
    flatDeepOrange: flatTheme(deepOrange),
    flatBrown: flatTheme(brown),
    flatGrey: flatTheme(grey),
    flatBlueGrey: flatTheme(blueGrey)
  };
};

export const _Button = ({ raised, classes, color, className, ...other }) => (
  <Button
    raised={raised}
    className={classNames(
      classes.root,
      {
        [raised ? classes.raisedRed : classes.flatRed]: color === 'red',
        [raised ? classes.raisedPink : classes.flatPink]: color === 'pink',
        [raised ? classes.raisedPurple : classes.flatPurple]: color === 'purple',
        [raised ? classes.raisedDeepPurple : classes.flatDeepPurple]: color === 'deepPurple',
        [raised ? classes.raisedIndigo : classes.flatIndigo]: color === 'indigo',
        [raised ? classes.raisedBlue : classes.flatBlue]: color === 'blue',
        [raised ? classes.raisedLightBlue : classes.flatLightBlue]: color === 'lightBlue',
        [raised ? classes.raisedCyan : classes.flatCyan]: color === 'cyan',
        [raised ? classes.raisedTeal : classes.flatTeal]: color === 'teal',
        [raised ? classes.raisedGreen : classes.flatGreen]: color === 'green',
        [raised ? classes.raisedLightGreen : classes.flatLightGreen]: color === 'lightGreen',
        [raised ? classes.raisedLime : classes.flatLime]: color === 'lime',
        [raised ? classes.raisedYellow : classes.flatYellow]: color === 'yellow',
        [raised ? classes.raisedAmber : classes.flatAmber]: color === 'amber',
        [raised ? classes.raisedOrange : classes.flatOrange]: color === 'orange',
        [raised ? classes.raisedDeepOrange : classes.flatDeepOrange]: color === 'deepOrange',
        [raised ? classes.raisedBrown : classes.flatBrown]: color === 'brown',
        [raised ? classes.raisedGrey : classes.flatGrey]: color === 'grey',
        [raised ? classes.raisedBlueGrey : classes.flatBlueGrey]: color === 'blueGrey'
      },
      className
    )}
    {...other}
  >
  </Button>
);

export default withStyles(styles)(_Button);
