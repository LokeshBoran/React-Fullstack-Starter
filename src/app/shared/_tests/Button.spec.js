import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import Button from '../Button';

configure({ adapter: new Adapter() });

describe('Button', () => {
  it('should render initial component', () => {
    const component = shallow(
      <Button />
    );

    expect(component).toMatchSnapshot();
  });
});
