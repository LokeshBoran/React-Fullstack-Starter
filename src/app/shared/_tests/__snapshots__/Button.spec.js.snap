// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render initial component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withStyles(Component) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "flatAmber": "Component-flatAmber-34",
        "flatBlue": "Component-flatBlue-26",
        "flatBlueGrey": "Component-flatBlueGrey-39",
        "flatBrown": "Component-flatBrown-37",
        "flatCyan": "Component-flatCyan-28",
        "flatDeepOrange": "Component-flatDeepOrange-36",
        "flatDeepPurple": "Component-flatDeepPurple-24",
        "flatGreen": "Component-flatGreen-30",
        "flatGrey": "Component-flatGrey-38",
        "flatIndigo": "Component-flatIndigo-25",
        "flatLightBlue": "Component-flatLightBlue-27",
        "flatLightGreen": "Component-flatLightGreen-31",
        "flatLime": "Component-flatLime-32",
        "flatOrange": "Component-flatOrange-35",
        "flatPink": "Component-flatPink-22",
        "flatPurple": "Component-flatPurple-23",
        "flatRed": "Component-flatRed-21",
        "flatTeal": "Component-flatTeal-29",
        "flatYellow": "Component-flatYellow-33",
        "raisedAmber": "Component-raisedAmber-15",
        "raisedBlue": "Component-raisedBlue-7",
        "raisedBlueGrey": "Component-raisedBlueGrey-20",
        "raisedBrown": "Component-raisedBrown-18",
        "raisedCyan": "Component-raisedCyan-9",
        "raisedDeepOrange": "Component-raisedDeepOrange-17",
        "raisedDeepPurple": "Component-raisedDeepPurple-5",
        "raisedGreen": "Component-raisedGreen-11",
        "raisedGrey": "Component-raisedGrey-19",
        "raisedIndigo": "Component-raisedIndigo-6",
        "raisedLightBlue": "Component-raisedLightBlue-8",
        "raisedLightGreen": "Component-raisedLightGreen-12",
        "raisedLime": "Component-raisedLime-13",
        "raisedOrange": "Component-raisedOrange-16",
        "raisedPink": "Component-raisedPink-3",
        "raisedPurple": "Component-raisedPurple-4",
        "raisedRed": "Component-raisedRed-2",
        "raisedTeal": "Component-raisedTeal-10",
        "raisedYellow": "Component-raisedYellow-14",
        "root": "Component-root-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "flatAmber": "Component-flatAmber-34",
          "flatBlue": "Component-flatBlue-26",
          "flatBlueGrey": "Component-flatBlueGrey-39",
          "flatBrown": "Component-flatBrown-37",
          "flatCyan": "Component-flatCyan-28",
          "flatDeepOrange": "Component-flatDeepOrange-36",
          "flatDeepPurple": "Component-flatDeepPurple-24",
          "flatGreen": "Component-flatGreen-30",
          "flatGrey": "Component-flatGrey-38",
          "flatIndigo": "Component-flatIndigo-25",
          "flatLightBlue": "Component-flatLightBlue-27",
          "flatLightGreen": "Component-flatLightGreen-31",
          "flatLime": "Component-flatLime-32",
          "flatOrange": "Component-flatOrange-35",
          "flatPink": "Component-flatPink-22",
          "flatPurple": "Component-flatPurple-23",
          "flatRed": "Component-flatRed-21",
          "flatTeal": "Component-flatTeal-29",
          "flatYellow": "Component-flatYellow-33",
          "raisedAmber": "Component-raisedAmber-15",
          "raisedBlue": "Component-raisedBlue-7",
          "raisedBlueGrey": "Component-raisedBlueGrey-20",
          "raisedBrown": "Component-raisedBrown-18",
          "raisedCyan": "Component-raisedCyan-9",
          "raisedDeepOrange": "Component-raisedDeepOrange-17",
          "raisedDeepPurple": "Component-raisedDeepPurple-5",
          "raisedGreen": "Component-raisedGreen-11",
          "raisedGrey": "Component-raisedGrey-19",
          "raisedIndigo": "Component-raisedIndigo-6",
          "raisedLightBlue": "Component-raisedLightBlue-8",
          "raisedLightGreen": "Component-raisedLightGreen-12",
          "raisedLime": "Component-raisedLime-13",
          "raisedOrange": "Component-raisedOrange-16",
          "raisedPink": "Component-raisedPink-3",
          "raisedPurple": "Component-raisedPurple-4",
          "raisedRed": "Component-raisedRed-2",
          "raisedTeal": "Component-raisedTeal-10",
          "raisedYellow": "Component-raisedYellow-14",
          "root": "Component-root-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
